core/storage/src/lib.rs

```diff
@@ -122,6 +124,58 @@ macro_rules! ensure_get {
     }};
 }
 
+macro_rules! get_keys_by_schema {
+    ($self_: ident, $vec: expr, $key_prefix: expr, $ibc_prefix: expr, $schema: ty) => {{
+        let prepare_iter = $self_.adapter.prepare_iter::<$schema, _>($key_prefix)?;
+        let mut res: Vec<LocalIbcPath> = prepare_iter
+            .ref_to_iter()
+            .filter_map(|pair| pair.ok().map(|(key, _)| key.0.try_into()))
+            .filter_map(|path: Result<LocalIbcPath, _>| {
+                path.ok().filter(|path| path.starts_with($ibc_prefix))
+            })
+            .collect();
+        $vec.append(&mut res);
+    }};
+}
```

Comment:
There is no need to use macros to simplify the code here. It is sufficient to use functions with generics.